### PR TITLE
Added support for RSSI & Link Quality warning settings

### PR DESCRIPTION
**Pull request by ianrmurphy**
Counterpart to https://github.com/betaflight/betaflight/pull/8017/.

Requires a target with MSP API v1.42.

### CODE REVIEW

File: src/js/tabs/osd.js [hunk 1]
```diff
@@ -1487,8 +1501,12 @@ OSD.msp = {
                 result.push16(warningFlags);
                 if (semver.gte(CONFIG.apiVersion, "1.41.0")) {
                     result.push32(warningFlags);
-
+                    // NOTE: betaflight MSP does not expect result.push8(OSD.data.osd_profiles.number);
                     result.push8(OSD.data.osd_profiles.selected + 1);
```
Comment (McGiverGim) [inline src/js/tabs/osd.js]: I don't understand this comment added. It does expect the number of current OSD profile here: https://github.com/betaflight/betaflight/blob/eb7e6c0b39695966bffc3849debe9ddcb700781d/src/main/msp/msp.c#L2758

File: src/js/tabs/osd.js [hunk 2]
```diff
@@ -1487,8 +1501,12 @@ OSD.msp = {
...
+                    result.push8(OSD.data.overlay_mode);
+                }
+                if (semver.gte(CONFIG.apiVersion, "1.42.0")) {
```
Comment (mikeller) [inline src/js/tabs/osd.js]: See https://github.com/betaflight/betaflight/pull/8017/files#r275298719

File: src/js/tabs/osd.js (hunk 1 above)
Comment (ianrmurphy) [reply src/js/tabs/osd.js]: MSP.c only expects `OSD.data.osd_profiles.selected` to be sent back from the Configurator, not `OSD.data.osd_profiles.number` which is implemented in Betaflight as the constant `OSD_PROFILE_COUNT`. I had a off-by-one byte alignment issue if I sent it, hence I added a comment to explain.

Anyway, this PR will need to wait until Betaflight 4.2 as 4.1 will have this feature as cli only.

Comment (mikeller) [reply src/js/tabs/osd.js]: @ianrmurphy: It will still need to have `RSSI` and `LINK_QUALITY` added to `ALL_WARNINGS`.

File: src/js/tabs/osd.js [hunk 3]
```diff
@@ -1107,8 +1107,15 @@ OSD.constants = {
         GPS_RESCUE_DISABLED: {
             name: 'GPS_RESCUE_DISABLED',
             desc: 'osdWarningGpsRescueDisabled'
-        }
-
+        },
+        RSSI: {
```
Comment (mikeller) [inline src/js/tabs/osd.js]: Eh, you got a bit too much now - the localised texts and the version check / adding of the warnings to `OSD.constants.WARNINGS` should stay as well - apologies for not being more explicit.

Comment (ianrmurphy) [reply src/js/tabs/osd.js]: No problem...

Comment (ianrmurphy) [reply src/js/tabs/osd.js]: Should I reference MSP 1.42 at all, or just add them back under 1.41?

Comment (mikeller) [reply src/js/tabs/osd.js]: 1.42 it is, as the firmware changes will go into 4.1

Comment (ianrmurphy) [reply src/js/tabs/osd.js]: Done.

**Pull request merged**